hazelcast/src/main/java/com/hazelcast/cache/impl/record/CacheDataRecord.java

```diff
@@ -26,9 +26,10 @@
 /**
  * Implementation of {@link com.hazelcast.cache.impl.record.CacheRecord} where value has an internal serialized format.
  */
-public class CacheDataRecord extends AbstractCacheRecord<Data> {
+public class CacheDataRecord extends AbstractCacheRecord<Data, Data> {
```

Comment:
We need to make sure that:
- either it is always serialized using `out.writeObject(instanceOfCacheDataRecord)`
- or it is serialized as part of an object graph with a `Versioned` root element and serialized via instanceOfCacheDataRecord.writeData(out)`

edit: is `CacheRecord` ever serialized at all?